File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/center.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Center — spec §2.11.
Widget buildCenter(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Center(
    widthFactor: parseDouble(p['widthFactor']),
    heightFactor: parseDouble(p['heightFactor']),
    child: r.maybeRender(context, node.child),
  );
}

File: docs/WidgetMill/packages/spec/src/nodes/center.ts
import { z } from "zod";
import { NodeRef } from "../node-ref";

/** Center — spec §2.11. */
export const CenterNode = z.object({
  type: z.literal("center"),
  props: z
    .object({
      widthFactor: z.number().optional(),
      heightFactor: z.number().optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});
